File: JWT/TokenExpiredException.cs

namespace JWT
{


    public class TokenExpiredException : System.Exception
    {
        public TokenExpiredException(string message)
            : base(message)
        {
        }
    }


    public class TokenAlgorithmRefusedException : System.Exception
    {
        public TokenAlgorithmRefusedException() 
            : this("Acceptance of the specified algorithm is denied.\nReason: Braindead JWT spec represents an unacceptable security risk.\n")
        {}

        public TokenAlgorithmRefusedException(string message)
            : base(message)
        {
        }
    }


    public class UnknownTokenAlgorithmException : System.Exception
    {
        public UnknownTokenAlgorithmException(string message)
            : base(message)
        {
        }
    }


}

File: vbBouncyJWT/TokenExpiredException.vb

Namespace BouncyJWT


    Public Class TokenExpiredException
        Inherits System.Exception
        Public Sub New(message As String)
            MyBase.New(message)
        End Sub
    End Class


    Public Class TokenAlgorithmRefusedException
        Inherits System.Exception
        Public Sub New()
            Me.New("Acceptance of the specified algorithm is denied." & vbLf & "Reason: Braindead JWT spec represents an unacceptable security risk." & vbLf)
        End Sub

        Public Sub New(message As String)
            MyBase.New(message)
        End Sub
    End Class


    Public Class UnknownTokenAlgorithmException
        Inherits System.Exception
        Public Sub New(message As String)
            MyBase.New(message)
        End Sub
    End Class


End Namespace
